src/Sylius/Bundle/ApiBundle/DataTransformer/ChannelPricingInputDataTransformer.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Sylius\Bundle\ApiBundle\DataTransformer;
+
+use ApiPlatform\Core\DataTransformer\DataTransformerInterface;
+use ApiPlatform\Core\Serializer\AbstractItemNormalizer;
+use Sylius\Bundle\ApiBundle\DTO\ChannelPricingInput;
+use Sylius\Bundle\ApiBundle\Helpers\PriceTransformer;
+use Sylius\Component\Core\Model\ChannelPricing;
+
+class ChannelPricingInputDataTransformer implements DataTransformerInterface
+{
+    /**
+     * @param object $input
+     * @param string $to
+     * @param array $context
+     * @return object|ChannelPricing
+     */
+    public function transform($input, string $to, array $context = [])
+    {
+        if (get_class($input) !== ChannelPricingInput::class) {
+            throw new \Exception('the $input parameter should be instance of ' . ChannelPricingInput::class);
+        }
+
+        if (($context[AbstractItemNormalizer::OBJECT_TO_POPULATE] ?? null)) {
+            $channelPricing = $context[AbstractItemNormalizer::OBJECT_TO_POPULATE];
+        } else {
+            $channelPricing = new ChannelPricing();
+        }
+
+        $channelPricing->setChannelCode($input->channelCode);
+        $channelPricing->setPrice(PriceTransformer::unitsToCentimes($input->price));
+        $channelPricing->setOriginalPrice(PriceTransformer::unitsToCentimes($input->originalPrice));
+
+        return $channelPricing;
+    }
+
+    public function supportsTransformation($data, string $to, array $context = []): bool
+    {
+        return $to === ChannelPricing::class;
+    }
```

Comment:
```suggestion
    public function supportsTransformation($data, string $to, array $context = []): bool
    {
        return $to === ChannelPricing::class && ChannelPricingInput::class === ($context['input']['class'] ?? null);
    }
```
But you have to check the context info. With this code all validation getting from one method

Reply:
https://github.com/bagrinsergiu/Sylius/pull/140#discussion_r730888602